src/main/java/org/datadog/jenkins/plugins/datadog/apm/RubyConfigurator.java

```diff
@@ -0,0 +1,37 @@
+package org.datadog.jenkins.plugins.datadog.apm;
+
+import hudson.FilePath;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import org.datadog.jenkins.plugins.datadog.steps.TestOptimization;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RubyConfigurator implements TracerConfigurator {
+    private static final int GET_VERSION_TIMEOUT_MILLIS = 30_000;
+    private static final int INSTALL_TRACER_TIMEOUT_MILLIS = 300_000;
+
+    @Override
+    public Map<String, String> configure(TestOptimization testOptimization, Node node, FilePath workspacePath, Map<String, String> envs, TaskListener listener) throws Exception {
+        String rubyVersion = workspacePath.act(new ShellCommandCallable(Collections.emptyMap(), GET_VERSION_TIMEOUT_MILLIS, "ruby", "-v"));
+        listener.getLogger().println("[datadog] Configuring DD Ruby tracer: got ruby version " + rubyVersion + " from " + workspacePath + " on " + node);
+
+        String bundlerVersion = workspacePath.act(new ShellCommandCallable(Collections.emptyMap(), GET_VERSION_TIMEOUT_MILLIS, "bundle", "-v"));
+        listener.getLogger().println("[datadog] Configuring DD Ruby tracer: got bundler version " + bundlerVersion + " from " + workspacePath + " on " + node);
+
+        String rubygemsVersion = workspacePath.act(new ShellCommandCallable(Collections.emptyMap(), GET_VERSION_TIMEOUT_MILLIS, "gem", "-v"));
+        listener.getLogger().println("[datadog] Configuring DD Ruby tracer: got rubygems version " + rubygemsVersion + " from " + workspacePath + " on " + node);
```

Comment:
Why are we not checking the versions that we get? Is the idea that as long as the command doesn't fail it should be enough?

Reply:
good point, in the script I check versions, here I forgot to do that because other Jenkins configurations don't check versions